src/gurt/telemetry.c

```diff
@@ -569,13 +572,16 @@ alloc_node(struct d_tm_shmem_hdr *shmem, struct d_tm_node_t **newnode,
 		rc = -DER_NO_SHMEM;
 		goto out;
 	}
-	rc = init_node(shmem, node, name);
+
+	tmp = conv_ptr(shmem, node);
```

Comment:
the return value is assigned to the sharemem struct directly somewhere, so it can blindly convert the ptr internally in shmalloc.  Though I agree the convertion is troublesome indeed, will try refining it in the following patch.